### PR TITLE
Integrate participant info form

### PR DESCRIPTION
**Pull request by jason17h**
### Notion ticket link

[Integrate Participant Information Form With Backend](https://www.notion.so/uwblueprintexecs/086c1b564dbf478faa649a8880e34b81?v=9d3f8002a0d1477ba395616dfc082af7&p=15c0ba7a02cb41ba8e8eb11e25bff1b9)

### Implementation description

- Allows users to enter information in the form fields and saves them to local storage (NOTE: there is currently a bug with rendering the checkboxes/radio buttons correctly on refresh using local storage - potentially a Chakra bug?)
- Upon completion of the form, the data is sent to the backend and saved in the database
- No frontend validation yet - only backend validation

### Steps to test

1. Checkout the branch
2. Login and fill out the form - ensure that you are authenticated!
3. Verify that the request was correctly sent via Postman

### What should reviewers focus on?

-

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: utils/validation/user.ts
Comment (feng-eric) [inline utils/validation/user.ts]:
```suggestion
        validationErrors.push("User first name is not alphanumeric");
```

File: utils/validation/user.ts
Comment (feng-eric) [inline utils/validation/user.ts]: What is the 2nd and 3rd parameter of `isAlpha`?

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]: @mahadzaryab1 mentioned that for typescript files, they should be camelCase so could you change it to `participantInfo.tsx`?

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]:
```suggestion
// TODO: Checkboxes have bugs in them; sometimes render sometimes don't
```

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]: Is this the placeholder in the Figma? Maybe you can ask @stacykwok @christinaruu what the text here should be.

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]: Why do we need to sort it?

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]: Do you need to specify this? If so, maybe we can add a util class for making requests similar to how we have `ResponseUtil` for the backend!

**Review by feng-eric — COMMENTED** [review]
Had some comments. I def do think we should split this page up but that can be done afterwards as a refactor task!

File: utils/validation/user.ts
Comment (jason17h) [reply utils/validation/user.ts]: 2nd parameter is the locale (defaults to `en-US`) and 3rd indicates which characters to allow - I actually used the 3rd param incorrectly here but I fixed it

File: pages/participantinfo.tsx
Comment (jason17h) [reply pages/participantinfo.tsx]: We don't, I just figured it would be easier to read in the database and since the list is very small it would come at a marginal cost. I could remove it though if you prefer?

File: pages/participantinfo.tsx
Comment (feng-eric) [reply pages/participantinfo.tsx:28]: @jason17h Bump! not sure if you saw this

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]:
```suggestion
                            placeholder="Vancouver"
```

File: pages/participantinfo.tsx
Comment (feng-eric) [inline pages/participantinfo.tsx]: Let's use this address as the placeholder values

1924 W 45th Ave, Vancouver, BC V6M 3S3

It's just a random elementary school I found in Vancouver lol

File: pages/participantinfo.tsx
Comment (feng-eric) [reply pages/participantinfo.tsx]: I think that's fine to keep

File: pages/participantinfo.tsx
Comment (jason17h) [reply pages/participantinfo.tsx]: Does the same thing apply to files in the `api` directory since these are also used as the API route? I tried changing it to `participantInfo` but the URL automatically changes back to `participantinfo` and it gives me this error page:
Screen Shot 2021-08-10 at 11 14 08 AM
Maybe using a dash would be better for readability? i.e., `participant-info.tsx`
(cc: @mahadzaryab1 )

File: pages/participantinfo.tsx
Comment (jason17h) [reply pages/participantinfo.tsx]: sure, or do you want me to use the actual SDC address?

File: pages/participantinfo.tsx
Comment (jason17h) [reply pages/participantinfo.tsx]: I think it is - when I exclude it I get a `500` error. Should I make that in this PR?

**Review by feng-eric — APPROVED** [review]
LGTM, let's get some more approvals before merging though

File: models/User.ts
Comment (mahadzaryab1) [inline models/User.ts]: it seems like the fields here are similar to the ones in `CreateStudentInput` - can we not just nest that type in here?

File: models/User.ts
Comment (mahadzaryab1) [inline models/User.ts]: same comment as above

File: prisma/schema.sql
Comment (mahadzaryab1) [inline prisma/schema.sql]: should this be snake_case?

File: prisma/schema.sql
Comment (mahadzaryab1) [inline prisma/schema.sql]: should this be snake_case?

**Review by mahadzaryab1 — COMMENTED** [review]
added a few comments/questions

File: prisma/schema.sql
Comment (feng-eric) [reply prisma/schema.sql]: Good catch, I can fix this one.

File: models/User.ts
Comment (feng-eric) [reply models/User.ts:46]: I can fix this one

File: prisma/schema.sql
Comment (feng-eric) [reply prisma/schema.sql]: Fixed this one.

File: models/User.ts
Comment (feng-eric) [reply models/User.ts]: I made the change [here](https://github.com/uwblueprint/social-diversity-for-children/pull/65/commits/dc7251ec6ca5096375c4080686ec84f0f8fc15e3). @mahadzaryab1 @jason17h @sohak22, could you guys take a look?

**Comment by jason17h** [comment]
I think I fixed the outstanding bugs so it looks good to me now! Thanks a bunch @feng-eric for helping to address the comments while I'm away

**Review by mahadzaryab1 — APPROVED** [review]
LGTM

**Pull request merged**
